cpg-core/src/test/kotlin/de/fraunhofer/aisec/cpg/passes/DFGTest.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.passes
+
+import de.fraunhofer.aisec.cpg.TestUtils
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.statements.ReturnStatement
+import java.nio.file.Path
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class DFGTest {
+    companion object {
+        private val topLevel = Path.of("src", "test", "resources", "dfg")
+    }
+    @Test
+    fun testReturnStatement() {
+        val result =
+            TestUtils.analyze(
+                listOf(Path.of(topLevel.toString(), "ReturnTest.java").toFile()),
+                topLevel,
+                true
+            )
+        val returnFunction = result.functions["testReturn"]
+        assertNotNull(returnFunction)
+
+        // TODO: For some reason, there's the "implicit" return statement even if the if and the
```

Comment:
I think the "implicit" return logic is rather stupid and only checks if the *last* statement of the compound statement is a return statement. we should probably filter for all returns first and then check if its necessary to include it.

Reply:
Here is the CPP code:
https://github.com/Fraunhofer-AISEC/cpg/blob/67522e366e549864a60f0c1b841f4c8a97a97bd6/cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclarationHandler.kt#L199-L212

Reply:
However, I can't see similar code for the java frontend (which you are using).

Reply:
Interesting. It seems that the JavaParser is inserting this `return`?!. If you look at the code of the `BlockStmt` given by JavaParser it is:
```java
{
    int a = 1;
    if (a == 5) {
        return 2;
    } else {
        return a;
    }
    return;
}
```

Even though the return is not part of the original source code.

Reply:
yeah, I was a bit confused because I didn't find the java-related code. But even if we have these statements, we do not have the EOG when inserting the return statement into the graph. So, I'm not sure if there's a good way to avoid adding these statements in the frontends.